providers/os/resources/packages/windows_packages.go

```diff
@@ -203,15 +214,90 @@ func (w *WinPkgManager) Format() string {
 	return "win"
 }
 
-const installedAppsScript = `
-Get-ItemProperty (@(
-  'HKLM:\\SOFTWARE\\Microsoft\\Windows\\CurrentVersion\\Uninstall\\*',
-  'HKCU:\\SOFTWARE\\Microsoft\\Windows\\CurrentVersion\\Uninstall\\*',
-  'HKLM:\\SOFTWARE\\Wow6432Node\\Microsoft\\Windows\\CurrentVersion\\Uninstall\\*',
-  'HKCU:\\SOFTWARE\\Wow6432Node\\Microsoft\\Windows\\CurrentVersion\\Uninstall\\*'
-) | Where-Object { Test-Path $_ }) |
-Select-Object -Property DisplayName,DisplayVersion,Publisher,EstimatedSize,InstallSource,UninstallString | ConvertTo-Json -Compress
-`
+func (w *WinPkgManager) getLocalInstalledApps() ([]Package, error) {
+	pkgs := []string{
+		"HKLM\\SOFTWARE\\Microsoft\\Windows\\CurrentVersion\\Uninstall",
+		"HKCU\\SOFTWARE\\Microsoft\\Windows\\CurrentVersion\\Uninstall",
+		"HKLM\\SOFTWARE\\Wow6432Node\\Microsoft\\Windows\\CurrentVersion\\Uninstall",
+		"HKCU\\SOFTWARE\\Wow6432Node\\Microsoft\\Windows\\CurrentVersion\\Uninstall",
+	}
+	packages := []Package{}
+	for _, r := range pkgs {
+		// we get all the packages, found under the pkgs paths
+		children, err := registry.GetNativeRegistryKeyChildren(r)
+		if err != nil {
+			continue
+		}
+		for _, c := range children {
+			// for each package the information is contained as children of that registry's key
+			p := w.getPackageFromRegistryKey(c)
+			if p == nil {
+				continue
+			}
+			packages = append(packages, *p)
+		}
+	}
+	return packages, nil
+}
+
+func (w *WinPkgManager) getInstalledApps() ([]Package, error) {
+	if w.conn.Type() == shared.Type_Local {
+		return w.getLocalInstalledApps()
+	}
+
+	cmd, err := w.conn.RunCommand(powershell.Wrap(installedAppsScript))
+	if err != nil {
+		return nil, fmt.Errorf("could not read app package list")
+	}
+	return ParseWindowsAppPackages(cmd.Stdout)
+}
+
+func (w *WinPkgManager) getPackageFromRegistryKey(key registry.RegistryKeyChild) *Package {
+	items, err := registry.GetNativeRegistryKeyItems(key.Path + "\\" + key.Name)
+	if err != nil {
+		log.Debug().Err(err).Str("path", key.Path).Str("name", key.Name).Msg("could not read registry key items")
+		return nil
+	}
+	var uninstallString string
+	var displayName string
+	var displayVersion string
+	var publisher string
+
+	for _, i := range items {
+		if i.Key == "UninstallString" {
+			uninstallString = i.Value.String
+		}
+		if i.Key == "DisplayName" {
+			displayName = i.Value.String
+		}
+		if i.Key == "DisplayVersion" {
+			displayVersion = i.Value.String
+		}
+		if i.Key == "Publisher" {
+			publisher = i.Value.String
+		}
+	}
+
+	if uninstallString == "" {
+		return nil
+	}
+
+	cpeWfn := ""
+	if displayName != "" && displayVersion != "" {
+		cpeWfn, err = cpe.NewPackage2Cpe(publisher, displayName, displayVersion, "", "")
+		if err != nil {
+			log.Debug().Err(err).Str("name", displayName).Str("version", displayVersion).Msg("could not create cpe for windows app package")
```

Comment:
is this CPE optional? or why we don't error here? 🤔

Reply:
I believe it is, yeah. Ive copied over the last implementation of how we fetch that.